internal/controller/services/monitoring/resources/opentelemetry-collector.tmpl.yaml

```diff
@@ -6,6 +6,22 @@ metadata:
 spec:
   replicas: {{.CollectorReplicas}}
   mode: deployment
+  resources:
+    limits:
+      cpu: {{.CollectorCPULimit}}
+      memory: {{.CollectorMemoryLimit}}
+    requests:
+      cpu: {{.CollectorCPURequest}}
+      memory: {{.CollectorMemoryRequest}}
+  volumeMounts:
+    - name: tls-certs
+      mountPath: /etc/otel-collector/tls
+      readOnly: true
+  volumes:
+    - name: tls-certs
+      secret:
+        secretName: data-science-collector-collector-headless-tls
+        optional: true
```

Comment:
Is this just from testing or does it also fix tls issue of collector? If I remember correctly this would work but only if a headless service was accessed, not the one we currently use.